File: frontend/src/app/domain/usecases/departments/update-department.usecase.ts
import { Injectable, inject } from '@angular/core';
import { Observable } from 'rxjs';
import { DepartmentRepository } from '@domain/repositories/department.repository';
import { Department } from '@domain/models/department.model';

@Injectable({ providedIn: 'root' })
export class UpdateDepartmentUseCase {
  private readonly repo = inject(DepartmentRepository);

  execute(id: string, name: string): Observable<Department> {
    const trimmed = name.trim();
    if (!trimmed) {
      throw new Error('Department name cannot be empty.');
    }
    return this.repo.update(id, trimmed);
  }
}

File: backend/modules/admin/application/departments/update_department_use_case.py
from modules.admin.domain.entities.department import Department
from modules.admin.domain.interfaces.repositories.i_department_repository import (
    IDepartmentRepository,
)
from modules.admin.domain.interfaces.use_cases.departments.i_update_department_use_case import (
    IUpdateDepartmentUseCase,
)


class UpdateDepartmentUseCase(IUpdateDepartmentUseCase):
    def __init__(self, repo: IDepartmentRepository) -> None:
        self._repo = repo

    async def execute(self, department_id: int, name: str) -> Department:
        return await self._repo.update(department_id, name)
